src/update-feeds/update-feeds.ts

```diff
@@ -89,12 +89,17 @@ export const runUpdateFeeds = async (providerName: ProviderName, chain: Chain, c
           dapisCount,
         };
       },
-      { totalTimeoutMs: dataFeedUpdateIntervalMs, attemptTimeoutMs: dataFeedUpdateIntervalMs }
+      { totalTimeoutMs: dataFeedUpdateIntervalMs }
     );
     if (!goFirstBatch.success) {
       logger.error(`Failed to get first active dAPIs batch`, goFirstBatch.error);
       return;
     }
+    if (Date.now() >= firstBatchStartTime + dataFeedUpdateIntervalMs) {
```

Comment:
It's still not bulletproof, but I am not sure we can actually implement this in a bulletproof way.